gammapy/spectrum/results.py

```diff
@@ -472,6 +472,19 @@ def __init__(self, **pars):
             setattr(self, k, v)
 
     @classmethod
+    def from_hap_output(cls, filename):
```

Comment:
if you mean this - I agree it's ugly but I am afraid it will have to stay until FitSpectrum serialization output is changed.
That also the reason why you don't see test for this. I don't think, we want FitSpectrum JSON files in gammapy-extra, do we?